commit.go

```diff
@@ -371,8 +369,8 @@ func (b *Builder) Commit(ctx context.Context, dest types.ImageReference, options
 		}
 	}
 
-	img, err := is.Transport.GetStoreImage(b.store, dest)
-	if err != nil && !errors.Is(err, storage.ErrImageUnknown) {
+	_, img, err := is.ResolveReference(dest)
+	if err != nil && err.Error() != fmt.Sprintf("trying to resolve a non-%s: reference %q", is.Transport.Name(), transports.ImageName(dest)) {
```

Comment:
This really should use `errors.{As,Is}`. `c/image/v5/storage.ErrNoSuchImage` should work.